osu.Game/Screens/Edit/Setup/FormBeatmapFileSelector.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Diagnostics;
+using System.IO;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Overlays;
+using osuTK;
+
+namespace osu.Game.Screens.Edit.Setup
+{
+    /// <summary>
+    /// A type of <see cref="FormFileSelector"/> dedicated to beatmap resources.
+    /// </summary>
+    /// <remarks>
+    /// This expands on <see cref="FormFileSelector"/> by adding an intermediate step before finalisation
+    /// to choose whether the selected file should be applied to the current difficulty or all difficulties in the set,
+    /// the user's choice is saved in <see cref="ApplyToAllDifficulties"/> before the file selection is finalised and propagated to <see cref="FormFileSelector.Current"/>.
+    /// </remarks>
+    public partial class FormBeatmapFileSelector : FormFileSelector
+    {
+        private readonly bool beatmapHasMultipleDifficulties;
+
+        public readonly Bindable<bool> ApplyToAllDifficulties = new Bindable<bool>(true);
+
+        public FormBeatmapFileSelector(bool beatmapHasMultipleDifficulties, params string[] handledExtensions)
+            : base(handledExtensions)
+        {
+            this.beatmapHasMultipleDifficulties = beatmapHasMultipleDifficulties;
+        }
+
+        protected override FileChooserPopover CreatePopover(string[] handledExtensions, Bindable<FileInfo?> current, string? chooserPath)
+        {
+            var popover = new BeatmapFileChooserPopover(handledExtensions, current, chooserPath, beatmapHasMultipleDifficulties);
+
+            popover.ApplyToAllDifficulties.ValueChanged += v =>
```

Comment:
What is this flow? Why isn't this a `BindTo` or something more standard?

Reply:
The idea was that the popover's bindable is nullable so that whatever the user chooses will always trigger a change in the bindable, but I don't want the general bindable to be nullable as well so there's a value changed event here.

Anyways I can understand why it looks odd, I've rewritten it in https://github.com/ppy/osu/pull/30860/commits/4a1401a33df7c3b489c6c0db05b68f6f1fe31079